src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -29,19 +32,134 @@ case class NoOpFingerprint() {
 }
 
 // IndexLogEntry-specific Content that uses IndexLogEntry-specific fingerprint.
-case class Content(root: String, directories: Seq[Content.Directory])
+case class Content(root: Directory) {
+
+  // List of fully qualified paths of all files mentioned in this Content object.
+  @JsonIgnore
+  lazy val files: Seq[Path] = {
+    // Recursively append parent dir path to subdirs and files.
```

Comment:
I think you are recursively finding files.

Reply:
yeah, listing files already present in the Content object => is recursive.
Adding files from a disk path to create a new Content object => not recursive as of now. (I can add it if it is required)

Reply:
Oh i see. let me fix it.

Reply:
thanks @imback82 , done